#! /bin/sh
patch -p1 -f $* < $0
exit $?

Marcus Brinkmann:

	* src/msgcache.c: Release hMapping.

Cherry picking some changes since the 3.0.0-rc2 release candidate:

2007-08-12 [wwp]	2.10.0cvs116

	* src/plugins/pgpcore/prefs_gpg.c
		Don't try to unset a GPG_AGENT_INFO that was not
		set (and don't use a NULL string in Windows, it was
		crashing with --debug if GPG_AGENT_INFO was not set).

2007-08-12 [wwp]	2.10.0cvs115

	* src/privacy.h
		Make gcc type-check arguments passed to privacy_set_error().

2007-08-12 [ticho]	2.10.0cvs114

	* src/prefs_folder_item.c
		Make sure that folder default account combobox always has some
		account preselected.
	* src/gtk/combobox.c
		Handle empty combobox gracefully (warning instead of crash).

2007-08-12 [ticho]	2.10.0cvs114

	* src/prefs_folder_item.c
		Make sure that folder default account combobox is not empty.
	* src/gtk/combobox.c
		Handle empty combobox gracefully (warning instead of crash).


diff -rup claws-mail-3.0.0-rc2/src/gtk/combobox.c claws-mail-2.10.0cvs116/src/gtk/combobox.c
--- claws-mail-3.0.0-rc2/src/gtk/combobox.c	2007-08-10 12:58:28.000000000 +0200
+++ claws-mail-2.10.0cvs116/src/gtk/combobox.c	2007-08-12 17:14:05.000000000 +0200
@@ -74,7 +74,7 @@ gint combobox_get_active_data(GtkComboBo
 
 	g_return_val_if_fail(combobox != NULL, -1);
 
-	gtk_combo_box_get_active_iter(combobox, &iter);
+	g_return_val_if_fail(gtk_combo_box_get_active_iter(combobox, &iter), -1);
 
 	model = gtk_combo_box_get_model(combobox);
 
diff -rup claws-mail-3.0.0-rc2/src/plugins/pgpcore/prefs_gpg.c claws-mail-2.10.0cvs116/src/plugins/pgpcore/prefs_gpg.c
--- claws-mail-3.0.0-rc2/src/plugins/pgpcore/prefs_gpg.c	2007-08-10 12:58:34.000000000 +0200
+++ claws-mail-2.10.0cvs116/src/plugins/pgpcore/prefs_gpg.c	2007-08-13 06:07:37.000000000 +0200
@@ -530,9 +530,13 @@ void prefs_gpg_enable_agent(gboolean ena
 			debug_print("Can't enable gpg agent (no GPG_AGENT_INFO)\n");
 		}
 	} else {
+		if (saved_gpg_agent_info) {
 			g_unsetenv("GPG_AGENT_INFO");
 			debug_print("unset GPG_AGENT_INFO=%s\n", 
 				saved_gpg_agent_info);
+		} else {
+			debug_print("Can't disable gpg agent (no GPG_AGENT_INFO)\n");
+		}
 	}
 }
 
diff -rup claws-mail-3.0.0-rc2/src/prefs_folder_item.c claws-mail-2.10.0cvs116/src/prefs_folder_item.c
--- claws-mail-3.0.0-rc2/src/prefs_folder_item.c	2007-08-10 12:58:30.000000000 +0200
+++ claws-mail-2.10.0cvs116/src/prefs_folder_item.c	2007-08-12 17:14:03.000000000 +0200
@@ -721,6 +721,7 @@ static void prefs_folder_item_compose_cr
 	GList *cur_ac;
 	GList *account_list;
 	PrefsAccount *ac_prefs;
+	gboolean default_account_set = FALSE;
 
 	page->item	   = item;
 
@@ -857,11 +858,20 @@ static void prefs_folder_item_compose_cr
 					ac_prefs->account_id);
 
 		/* Set combobox to current default account id */
-		if (ac_prefs->account_id == item->prefs->default_account)
+		if (ac_prefs->account_id == item->prefs->default_account) {
 			combobox_select_by_data(GTK_COMBO_BOX(optmenu_default_account),
 					ac_prefs->account_id);
+			default_account_set = TRUE;
+		}
 	}
 
+	/* If nothing has been set (folder doesn't have a default account set),
+	 * pre-select global default account, since that's what actually used
+	 * anyway. We don't want nothing selected in combobox. */
+	if( !default_account_set )
+		combobox_select_by_data(GTK_COMBO_BOX(optmenu_default_account),
+				account_get_default()->account_id);
+
 	SET_TOGGLE_SENSITIVITY(checkbtn_enable_default_account, optmenu_default_account);
 
 	default_account_rec_checkbtn = gtk_check_button_new();
diff -rup claws-mail-3.0.0-rc2/src/privacy.h claws-mail-2.10.0cvs116/src/privacy.h
--- claws-mail-3.0.0-rc2/src/privacy.h	2007-08-10 12:58:29.000000000 +0200
+++ claws-mail-2.10.0cvs116/src/privacy.h	2007-08-12 17:14:03.000000000 +0200
@@ -66,7 +66,7 @@ gboolean privacy_encrypt			(const gchar 
 						 MimeInfo     *mimeinfo,
 						 const gchar  *encdata);
 
-void privacy_set_error				(const gchar *format, ...);
+void privacy_set_error				(const gchar *format, ...) G_GNUC_PRINTF(1, 2);
 void privacy_reset_error				(void);
 gboolean privacy_peek_error				(void);
 const gchar *privacy_get_error			(void);


--- claws-mail-old/src/msgcache.c	2007-08-13 15:42:39.000000000 +0200
+++ claws-mail/src/msgcache.c	2007-08-13 15:43:26.000000000 +0200
@@ -648,6 +648,7 @@
 			if (!hMapping)
 				goto w32_fail;
 			cache_data = (unsigned char *)MapViewOfFile(hMapping, FILE_MAP_COPY, 0, 0, 0);
+			CloseHandle (hMapping);
 		w32_fail:
 			;
 #else
@@ -830,6 +831,7 @@
 			if (!hMapping)
 				goto w32_fail2;
 			cache_data = (unsigned char *)MapViewOfFile(hMapping, FILE_MAP_COPY, 0, 0, 0);
+			CloseHandle (hMapping);
 		w32_fail2:
 			;
 #else
@@ -913,6 +915,7 @@
 			if (!hMapping)
 				goto w32_fail6;
 			cache_data = (unsigned char *)MapViewOfFile(hMapping, FILE_MAP_COPY, 0, 0, 0);
+			CloseHandle (hMapping);
 		w32_fail6:
 			;
 #else
@@ -1248,6 +1251,7 @@
 			if (!hMapping)
 				goto w32_fail3;
 			cache_data = (unsigned char *)MapViewOfFile(hMapping, FILE_MAP_COPY, 0, 0, 0);
+			CloseHandle (hMapping);
 		w32_fail3:
 			;
 #else
@@ -1267,6 +1271,7 @@
 				if (!hMapping)
 					goto w32_fail4;
 				mark_data = (unsigned char *)MapViewOfFile(hMapping, FILE_MAP_COPY, 0, 0, 0);
+				CloseHandle (hMapping);
 			w32_fail4:
 				;
 #else
@@ -1293,6 +1298,7 @@
 						if (!hMapping)
 							goto w32_fail5;
 						tags_data = (unsigned char *)MapViewOfFile(hMapping, FILE_MAP_COPY, 0, 0, 0);
+						CloseHandle (hMapping);
 					w32_fail5:
 						;
 #else
